### PR TITLE
feat(markdown-docx): add thematic-break transformer - #397

### PR DESCRIPTION
**Pull request by K-Kumar-01**
Signed-off-by: K-Kumar-01 <kushalkumargupta4@gmail.com>

### Changes

- Transformer(CiceroMark<->OOXML)
- Tests for the same

### Screenshots or Video

95ac184e-7258-40e4-90fd-20b5e77a4520

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [x] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `master` from `fork:branchname`

### CODE REVIEW

File: packages/markdown-docx/src/ToOOXMLVisitor/rules.js [hunk 1]
```diff
@@ -169,6 +169,18 @@ const CODE_PROPERTIES_RULE = () => {
     `;
 };
 
+const THEMATICBREAK_RULE = () => {
+    return `
+      <w:p>
+        <w:pPr>
+          <w:pBdr>
+            <w:bottom w:val="single" w:sz="12" w:space="1" w:color="auto"/>
```
Comment (algomaster99) [inline packages/markdown-docx/src/ToOOXMLVisitor/rules.js:177]: Do we need the offset (`w:space`) value? [1 point offset is equivalent to 1/72th of an inch](http://officeopenxml.com/WPborders.php) so I don't think it makes much of a difference.

**Review by algomaster99 — CHANGES_REQUESTED** [review]
Minor comments. If some things are not possible, let me know. We can proceed for merge anyway.

File: packages/markdown-docx/src/ToOOXMLVisitor/rules.js (hunk 1 above)
Comment (K-Kumar-01) [reply packages/markdown-docx/src/ToOOXMLVisitor/rules.js:177]: @algomaster99 
Does the first one seems fine?
78f414bf-1e30-4c3e-9309-7cb28989fca9

Comment (algomaster99) [reply packages/markdown-docx/src/ToOOXMLVisitor/rules.js:177]: The last one is okay. It just needs some offset.

**Comment by K-Kumar-01** [comment]
@algomaster99 I haven't changed the `w:space` property. 
Let me know if there are any changes required.

**Comment by K-Kumar-01** [comment]
@algomaster99 
requesting your review

**Comment by K-Kumar-01** [comment]
@algomaster99 
I have pushed the changes as discussed in the meeting.
A point to be noted- The border `thickness` is the minimum. I tried changing `w:sz` to `1` and `3` but it didn't have any effect on the above.

**Comment by algomaster99** [comment]
@K-Kumar-01 Did you try using the `between` attribute which we discovered during the meeting?

**Comment by K-Kumar-01** [comment]
> @K-Kumar-01 Did you try using the `between` attribute which we discovered during the meeting?

@algomaster99 Yes I did try but I did not like it. The `between` attribute was being given to all paragraphs for which the line between them was needed.

**Comment by K-Kumar-01** [comment]
@algomaster99 
Can we merge this one?

**Review by algomaster99 — COMMENTED** [review]
Looks good to me! You can also add cases for the following CiceroMark and then we would be good to go.

**1st case**
```
{
  "$class": "org.accordproject.commonmark.Document",
  "xmlns": "http://commonmark.org/xml/1.0",
  "nodes": [
    {
      "$class": "org.accordproject.templatemark.ContractDefinition",
      "name": "top",
      "nodes": [
        {
          "$class": "org.accordproject.commonmark.Paragraph",
          "nodes": [
            {
              "$class": "org.accordproject.commonmark.Text",
              "text": "First paragraph"
            }
          ]
        },
        {
          "$class": "org.accordproject.commonmark.ThematicBreak"
        },
        {
          "$class": "org.accordproject.commonmark.Paragraph",
          "nodes": [
            {
              "$class": "org.accordproject.commonmark.Text",
              "text": "Second paragraph"
            }
          ]
        },
        {
          "$class": "org.accordproject.commonmark.ThematicBreak"
        },
        {
          "$class": "org.accordproject.commonmark.Paragraph",
          "nodes": [
            {
              "$class": "org.accordproject.commonmark.Text",
              "text": "Third paragraph"
            }
          ]
        }
      ]
    }
  ]
}
```
**2nd case**
```
{
  "$class": "org.accordproject.commonmark.Document",
  "xmlns": "http://commonmark.org/xml/1.0",
  "nodes": [
    {
      "$class": "org.accordproject.templatemark.ContractDefinition",
      "name": "top",
      "nodes": [
        {
          "$class": "org.accordproject.commonmark.Paragraph",
          "nodes": [
            {
              "$class": "org.accordproject.commonmark.Text",
              "text": "First paragraph"
            }
          ]
        },
        {
          "$class": "org.accordproject.commonmark.ThematicBreak"
        },
        {
          "$class": "org.accordproject.commonmark.ThematicBreak"
        },
        {
          "$class": "org.accordproject.commonmark.Paragraph",
          "nodes": [
            {
              "$class": "org.accordproject.commonmark.Text",
              "text": "Third paragraph"
            }
          ]
        }
      ]
    }
  ]
}
```

**Comment by K-Kumar-01** [comment]
@algomaster99 
Added the test cases.

**Comment by algomaster99** [comment]
@K-Kumar-01 the 2nd test case is not rendering well on MS Word. It somehow does not render the first thematic break.
render-error

Could you look at it?

**Comment by K-Kumar-01** [comment]
> @K-Kumar-01 the 2nd test case is not rendering well on MS Word. It somehow does not render the first thematic break.
> render-error
> 
> Could you look at it?

Yeah, I had a look at it and it seems off.
Continuous thematic breaks seem to have `top`,` bottom` borders. More than two continuous segments seem to have a `between` property between them.

However, by doing some changes we can achieve the desired result.
Screenshot (85)

Change made:
Remove the `w:spacing`  tag and add `<w:p/>` tag.

I am not sure the exact reason as to why there is a need for such but it is easier than all other methods.

Other methods:
Check how many continuous horizontal rules are there and then use them to generate paragraphs. However, this itself is a complex task.

I am in favour of the first method as it is quite easy to implement. 😬

**Comment by algomaster99** [comment]
> Remove the w:spacing tag and add <w:p/> tag.

Push this and I will see it once. This PR is quite close to getting merge so I don't think so there will be much work.

> Check how many continuous horizontal rules are there and then use them to generate paragraphs. However, this itself is a complex task.

Yeah, this is onerous.

**Review by algomaster99 — APPROVED** [review]

**Pull request merged**